database/events/__init__.py

```diff
@@ -126,7 +130,8 @@ def registration(event_id):
             validate(data, event.registration_schema)
         except ValidationError as e:
             flash(e.message, 'danger')
-            return render_template('events/registration.html', form=form, event=event)
+            return render_template('events/registration.html', form=form,
+                                   event=event, registration=None)
```

Comment:
What does adding `registration=None` change here? Aren't all variables not defined in jinja templates `None`?

Reply:
True... and if I think of this, the problem is not really solved. Good pick.